core/src/main/java/jenkins/ClassLoaderReflectionToolkit.java

```diff
@@ -59,52 +36,121 @@
     }
 
     private static Object getClassLoadingLock(ClassLoader cl, String name) {
+        initClassLoadingLock();
+
         return invoke(GET_CLASS_LOADING_LOCK, RuntimeException.class, cl, name);
     }
 
+    private static void initClassLoadingLock() {
+        if (GET_CLASS_LOADING_LOCK == null) {
+            Method gCLL;
+            try {
+                gCLL = ClassLoader.class.getDeclaredMethod("getClassLoadingLock", String.class);
+                gCLL.setAccessible(true);
+            } catch (NoSuchMethodException x) {
+                throw new AssertionError(x);
+            }
+            GET_CLASS_LOADING_LOCK = gCLL;
```

Comment:
Not thread-safe. You could either make this and similar methods `synchonized`, or use a `private static` nested holder class with `final` variables to implement lazy initialization (which would make for a simpler patch overall I guess).

Reply:
second method sounds good :+1:

thanks I was worried about that bit